ravenframework/Optimizers/SimulatedAnnealing.py

```diff
@@ -694,23 +723,29 @@ def _nextNeighbour(self, rlz, fraction=1):
     """
     nextNeighbour = {}
     D = len(self.toBeSampled.keys())
-    alpha = 0.94
+    learnRate = self._coolingParameters.get('learningRate', 0.9)
     if self._coolingMethod in ['exponential', 'geometric']:
-      amp = ((fraction)**-1) / 20
+      amp = (fraction ** -1) / 20.0 # aa: I don't find a reference that justifies this factor 20.0
       r = randomUtils.random(dim=D, samples=1)
       delta = (-amp/2.)+ amp * r
     elif self._coolingMethod == 'boltzmann':
-      amp = min(np.sqrt(self.T), 1/3.0/alpha)
-      delta =  randomUtils.randomNormal(size=D)*alpha*amp
+      amp = [min(np.sqrt(self.T), 1.0/3.0/learnRate) for var in self.toBeSampled.keys()]
+      delta = np.asarray(amp).flatten() * randomUtils.randomNormal(size=(D,)).flatten() * learnRate
     elif self._coolingMethod == 'veryfast':
       amp = randomUtils.random(dim=D, samples=1)
       delta = np.sign(amp-0.5)*self.T*((1+1.0/self.T)**abs(2*amp-1)-1.0)
     elif self._coolingMethod == 'cauchy':
-      amp = (np.pi - (-np.pi))*randomUtils.random(dim=D, samples=1)-np.pi
-      delta = alpha*self.T*np.tan(amp)
+      amp = 2 * np.pi*randomUtils.random(dim=D, samples=1)-(np.pi)
+      delta = learnRate*self.T*np.tan(amp)
     delta = np.atleast_1d(delta)
+
     for i,var in enumerate(self.toBeSampled.keys()):
-      nextNeighbour[var] = rlz[var] + delta[i]
+      nextNeighbour[var] = rlzNormalized[var] + delta[i]
+      nextNeighbour[var] =  0 if nextNeighbour[var] < 0 else nextNeighbour[var]
+      nextNeighbour[var] =  1 if nextNeighbour[var] > 1 else nextNeighbour[var]
+      if self.distDict[var].distType == distType.discrete:
+        val = nextNeighbour[var]
+        nextNeighbour[var] = self.distDict[var].cdf(self.distDict[var].ppf(val))
```

Comment:
Do you need to denormalize the data before you compute ppf?

Reply:
Theoretically the normalization is performed between 0-1 and it implicitly maps to a uniform distribution. I can add the denormalization in order to avoid that, if in the future the normalization strategy got changed, this would result in a diff

Reply:
added